src/subclass/icon_view.rs

```diff
@@ -1,13 +1,252 @@
+use glib_sys;
+use gtk_sys;
+
+use libc::c_int;
+
 use glib::subclass::prelude::*;
+use glib::translate::*;
 
 use super::container::ContainerImpl;
 use ContainerClass;
+use IconView;
 use IconViewClass;
+use MovementStep;
+use TreePath;
+
+pub trait IconViewImpl: IconViewImplExt + ContainerImpl + 'static {
+    fn item_activated(&self, icon_view: &IconView, path: &mut TreePath) {
```

Comment:
This should probably not be a mutable reference, or is this supposed to be changeable by the subclass? It maps to a signal and to a normal method, the former doesn't mention anything about it being modifiable, the latter takes an immutable reference.

Reply:
I had no choice. The underlying vfunc in the class expect a `*mut` and it seems it's the only way.

`    pub item_activated: Option<unsafe extern "C" fn(*mut GtkIconView, *mut GtkTreePath)>,`

Reply:
Note that it `&IconView` even it `*mut GtkIconView` 😉 
So better check docs and possible just remove `mut`

Reply:
`glib::translate::mut_override()` to go from `*const T` to `*mut T`. That's generally caused by C API not using `const` correctly, so all pointers are `*mut` on the Rust side.

Reply:
@sdroege Strange, but seems this function is only one that not initialized by default even if gtk4
https://github.com/GNOME/gtk/blob/mainline/gtk/gtkiconview.c#L369
https://github.com/GNOME/gtk/blob/mainline/gtk/gtkiconviewprivate.h#L55

Reply:
Yeah, just doing nothing if not implemented by a subclass.

Reply:
ok I use `mut_override()` here. And removed mut